specification/cognitiveservices/data-plane/TranslatorText/preview/v1.0-preview.1/TranslatorBatch.json

```diff
@@ -117,22 +117,61 @@
+            "description": "$top indicates the total number of records the user wants to be returned across all pages.\r\n            \r\nClients MAY use $top and $skip query parameters to specify a number of results to return and an offset into the collection.\r\nWhen both $top and $skip are given by a client, the server SHOULD first apply $skip and then $top on the collection.\r\n            \r\nNote: If the server can't honor $top and/or $skip, the server MUST return an error to the client informing about it instead of just ignoring the query options.",
             "type": "integer",
             "format": "int32",
-            "default": 50,
-            "maximum": 100,
-            "minimum": 1
+            "maximum": 2147483647,
+            "minimum": 0
           },
           {
             "in": "query",
             "name": "$skip",
-            "description": "Skip the $skip entries in the collection\r\nWhen both $top and $skip are supplied, $skip is applied first",
+            "description": "$skip indicates the number of records to skip from the list of records held by the server based on the sorting method specified.  By default, we sort by descending start time.\r\n            \r\nClients MAY use $top and $skip query parameters to specify a number of results to return and an offset into the collection.\r\nWhen both $top and $skip are given by a client, the server SHOULD first apply $skip and then $top on the collection.\r\n            \r\nNote: If the server can't honor $top and/or $skip, the server MUST return an error to the client informing about it instead of just ignoring the query options.",
             "type": "integer",
             "format": "int32",
             "default": 0,
             "maximum": 2147483647,
             "minimum": 0
+          },
+          {
+            "in": "query",
+            "name": "$maxpagesize",
+            "description": "$maxpagesize is the maximum items returned in the items in a page.  If more items are requested via $top (or $top is not specified and there are more items to be returned), @nextLink will contain the link to the next page.\r\n            \r\nClients MAY request server-driven paging with a specific page size by specifying a $maxpagesize preference. The server SHOULD honor this preference if the specified page size is smaller than the server's default page size.",
+            "type": "integer",
+            "format": "int32",
+            "default": 50,
+            "maximum": 100,
+            "minimum": 1
+          },
+          {
+            "in": "query",
+            "name": "ids",
+            "description": "Ids to use in filtering",
+            "type": "string"
+          },
+          {
+            "in": "query",
+            "name": "status",
+            "description": "Status to use in filtering",
+            "type": "string"
```

Comment:
Wondering if all possible status string options should be added here in an enum. Kind of like how Form Recognizer does it: https://github.com/Azure/azure-rest-api-specs/blob/master/specification/cognitiveservices/data-plane/FormRecognizer/preview/v2.1-preview.2/FormRecognizer.json#L2141-L2151

@maririos

Reply:
it's now an enum type

Reply:
hmm, swagger gen failed for this...will take a look more.

Reply:
wonder if this is the issue... https://github.com/domaindrivendev/Swashbuckle.AspNetCore/issues/2034

Reply:
looks like it's not possible based on swagger 2.0 spec

https://swagger.io/specification/v2/


 The type of the parameter. Since the parameter is not located at the request body, it is limited to simple types (that is, not an object). The value MUST be one of "string", "number", "integer", "boolean", "array" or "file". If type is "file", the consumes MUST be either "multipart/form-data", " application/x-www-form-urlencoded" or both and the parameter MUST be in "formData".